src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs

```diff
@@ -173,47 +174,66 @@ public override int GetMatsOsPlatformCode()
 
         public override Task StartDefaultOsBrowserAsync(string url)
         {
-            try
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                var psi = new ProcessStartInfo
+                try
                 {
-                    FileName = url,
-                    UseShellExecute = true
-                };
-                Process.Start(psi);
-            }
-            catch
-            {
-                // hack because of this: https://github.com/dotnet/corefx/issues/10361
-                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    var psi = new ProcessStartInfo
+                    {
+                        FileName = url,
+                        UseShellExecute = true
+                    };
+                    Process.Start(psi);
+                }
+                catch
                 {
+                    // hack because of this: https://github.com/dotnet/corefx/issues/10361
                     url = url.Replace("&", "^&");
                     Process.Start(new ProcessStartInfo("cmd", $"/c start msedge {url}") { CreateNoWindow = true });
                 }
-                else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```

Comment:
nit: use the helpers here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/b3d83bdefab13ff7a2313aad6434fb34da210e8e/src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOs/DesktopOsHelper.cs#L92

Reply:
Done.